web_fragments/tests/test_fragment.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Unit tests for the Fragment class.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import ddt
+
+from django.test import TestCase
+
+from web_fragments.fragment import Fragment, FragmentResource
+from web_fragments.test_utils import (
+    CSS_EXPECTED_HTML, CSS_LINK_EXPECTED_HTML, EXPECTED_JS_INIT_VERSION, JS_EXPECTED_HTML,
+    JS_LINK_EXPECTED_HTML, TEST_CSS, TEST_CSS_URL, TEST_HTML, TEST_JS, TEST_JS_INIT_FN,
+    TEST_JS_URL, TEST_JSON_INIT_ARGS
+)
+
+
+@ddt.ddt
+class TestFragment(TestCase):
+    """
+    Unit tests for fragments.
+    """
+
+    def setUp(self):
+        super(TestFragment, self).setUp()
+
+    def create_test_fragment(self):
+        """
+        Creates a fragment for use in unit tests.
+        """
+        fragment = Fragment()
+        fragment.add_content(TEST_HTML)
+        fragment.add_css(TEST_CSS)
+        fragment.add_css_url(TEST_CSS_URL)
+        fragment.add_javascript(TEST_JS)
+        fragment.add_javascript_url(TEST_JS_URL)
+        fragment.initialize_js(TEST_JS_INIT_FN, json_args=TEST_JSON_INIT_ARGS)
+        return fragment
+
+    def validate_fragment(self, fragment=None, fragment_dict=None):
+        """
+        Validates that the fields of a fragment are all correct.
+        """
+        fragment_dict = fragment_dict if fragment_dict else fragment.to_dict()
+        self.assertEqual(fragment_dict['content'], TEST_HTML)
```

Comment:
@andy-armstrong Global comment for all test code: `pytest` is much smarter when it comes to asserting on something, so you might want to stick with the pytest way: http://doc.pytest.org/en/latest/assert.html

I.e. just do `assert fragment_dict['content'] == TEST_HTML`